tests/common/apiversion.go

```diff
@@ -0,0 +1,58 @@
+package common
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+var (
+	ingressPath = "/01_keycloak_helmchart/keycloak/templates/ingress.yaml"
+)
+
+func IngressApiVersionTest(t *testing.T, basepath string) {
+	// Setup decoding
+	subject := &networkingv1.Ingress{}
+	scheme := NewSchemeWithDefault(t)
+	require.NoError(t, networkingv1.AddToScheme(scheme))
+
+	// Read file
+	data, err := ioutil.ReadFile(basepath + ingressPath)
+	require.NoError(t, err)
+
+	// Decode object
+	kind := subject.GetObjectKind().GroupVersionKind()
+	decode, _, err := serializer.NewCodecFactory(scheme).UniversalDeserializer().Decode(data, &kind, subject)
+	require.NoError(t, err)
+	ingress, ok := decode.(*networkingv1.Ingress)
+	assert.True(t, ok, "Expected networking.k8s.io/v1 Ingress object")
+	if ok {
+		assert.Equal(t, "keycloak", ingress.Name)
+	}
+}
+
+func StatefulSetApiVersionTest(t *testing.T, path, expectedName string) {
+	// Setup decoding
+	subject := &appsv1.StatefulSet{}
+	scheme := NewSchemeWithDefault(t)
+	require.NoError(t, appsv1.AddToScheme(scheme))
+
+	// Read file
+	data, err := ioutil.ReadFile(path+"/statefulset.yaml")
+	require.NoError(t, err)
+
+	// Decode object
+	kind := subject.GetObjectKind().GroupVersionKind()
+	decode, _, err := serializer.NewCodecFactory(scheme).UniversalDeserializer().Decode(data, &kind, subject)
+	require.NoError(t, err)
+	sts, ok := decode.(*appsv1.StatefulSet)
+	assert.True(t, ok, "Expected apps/v1 Ingress object")
+	if ok {
+		assert.Equal(t, expectedName, sts.Name)
+	}
+}
```

Comment:
Why is this test necessary? There are already tests that load/decode statefulsets. In the `common` boilerplate we have already the necessary decoding. You should be able to use this simplified version:
```suggestion
func StatefulSetApiVersionTest(t *testing.T, path, expectedName string) {
	subject := DecodeStatefulsetV1(t, path+"/statefulset.yaml")
	assert.Equal(t, expectedName, subject.Name)
}
```

Reply:
I don't want a raw panic if the apiVersion is wrong, which is why I replicated the code from the common boilerplate here.